internal/services/keyvault/key_vault_key_resource.go

```diff
@@ -139,6 +141,62 @@ func resourceKeyVaultKey() *pluginsdk.Resource {
 				ValidateFunc: validation.IsRFC3339Time,
 			},
 
+			"rotation_policy": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						// at least 28 days
+						"expiry_time": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: validate.ISO8601Duration,
+							AtLeastOneOf: []string{
+								"rotation_policy.0.expiry_time",
+								"rotation_policy.0.auto_rotation",
+							},
+						},
+
+						// >= expiry_time - 7
+						"notification_time": {
```

Comment:
also a duration? should be notify_after? notify_before?

Reply:
Going for `notify_before_expiry`